modules/exploits/multi/http/navigate_cms_rce.rb

```diff
@@ -0,0 +1,116 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Navigate CMS Unauthenticated Remote Code Execution',
+      'Description' => %q(
+      This module exploits insufficient sanitization in the database::protect
+      method to bypass authentication.
+      It then uses a path traversal vulnerability in navigate_upload.php that
+      allows authenticated users to upload PHP files to arbitrary locations.
+      Together these vulnerabilities allow an unauthenticated attacker to
+      execute arbitrary PHP code remotely.
+
+      This module was tested against Navigate CMS 2.8.
+      ),
+      'Author' =>
+        [
+          'Pyriphlegethon' # Discovery / msf module
+        ],
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['CVE', '2018-17552'], # Authentication bypass
+          ['CVE', '2018-17553']  # File upload
+        ],
+      'Privileged' => false,
+      'Platform' => ['php'],
+      'Arch' => ARCH_PHP,
+      'Payload' => {},
+      'Targets' =>
+        [
+          ['Automatic', {}]
+        ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Sep 26 2018'))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base Navigate CMS directory path', '/navigate/'])
+      ]
+    )
+  end
+
+  def check
+    check_resp = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/login.php')
+    )
+
+    if check_resp && check_resp.body.include?('Navigate CMS v2.8')
```

Comment:
Are versions prior to 2.8 vulnerable?

Reply:
`CheckCode::Vulnerable` or `CheckCode::Appears` seems appropriate, if you can confirm the auth bypass.

The module description should also be updated. If you don't know the specific version, then consider using vague wording. Also, the module description should also mention the product name and vulnerable versions, so something like:

```
This module exploits insufficient sanitization in the database::protect method,
of NavigateCMS versions 2.8.5 and prior, to bypass authentication.

It then uses a path traversal vulnerability in navigate_upload.php that allows
authenticated users to upload PHP files to arbitrary locations. Together these
vulnerabilities allow an unauthenticated attacker to execute arbitrary PHP
code remotely.

This module was tested against Navigate CMS 2.8.
```